sechub-commons-model/src/main/java/com/mercedesbenz/sechub/commons/communication/CommunicationDataConverter.java

```diff
@@ -0,0 +1,114 @@
+package com.mercedesbenz.sechub.commons.communication;
+
+import static java.util.Objects.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import com.mercedesbenz.sechub.commons.communication.CommunicationDataConverterConfig.Receive;
+import com.mercedesbenz.sechub.commons.communication.CommunicationDataConverterConfig.Send;
+import com.mercedesbenz.sechub.commons.model.JSONConverter;
+
+/**
+ * Converts data for communication between sending and receiving. The way of
+ * conversion can be configured by configuration object.
+ *
+ * A usage example:
+ *
+ * <pre>
+ * <code>
+ * CommunicationDataConverter converter = new CommunicationDataConverter(config);
+ *
+ * val requestData = new HashMap<String,String>();
+ * requestData.put("cweId", finding.getCweId());
+ * requestData.put("language", finding.getLanguage());
+ * requestData.put("snippet", finding.getSnippet());
+ *
+ * String jsonToSend = converter.convertForSending(data);
+ *
+ * String jsonReceived = issueDescriberAiService.describe(jsonToSend);
+ *
+ * Map<String, String> resultMappingData= converter.convertFromReceived(jsonReceived);
+ * String failed = resultMappingData.get("ai_service_failure")
+ * String descriptionAsMarkdown = resultMappingData.get("content_markdown");
+ * // ...
+ * </code>
+ * </pre>
+ *
+ *
+ */
+public class CommunicationDataConverter {
+
+    private CommunicationDataConverterConfig config;
+
+    /**
+     * Creates the converter
+     *
+     * @param config the configuration to user, may not be <code>null</code>
+     *
+     * @throws IllegalArgumentException when config is <code>null</code>
```

Comment:
The Javadoc `throws` does not fit the methods behaviour.
```suggestion
     * @throws IllegalArgumentException when config send or receive CommunicationDataConversionType is not supported
```